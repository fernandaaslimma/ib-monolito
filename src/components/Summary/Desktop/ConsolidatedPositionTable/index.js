import { connect } from "redux-zero/react";

import ConsolidatedPositionTable from "./ConsolidatedPositionTable";
import actions from "../../../../actions/summary";

export default connect(
  ({ assets, currentIndex, getAssetsByType, loading }) => ({
    assets,
    currentIndex,
    getAssetsByType,
    loading
  }),
  actions
)(ConsolidatedPositionTable);
