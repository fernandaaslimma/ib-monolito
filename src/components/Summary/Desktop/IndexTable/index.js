import IndexTable from "./IndexTable";

export default IndexTable;
