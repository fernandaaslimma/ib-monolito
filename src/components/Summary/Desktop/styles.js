import styled, { css } from "styled-components";
import { rem, media } from "../../../styles/tools";
import { white30 } from "../../../styles/settings";

export const Wrapper = styled.div`
  background: ${white30};
  padding: ${rem(15)} 0;

  ${media.md(css`
    padding-top: ${rem(27)};
  `)};
`;
