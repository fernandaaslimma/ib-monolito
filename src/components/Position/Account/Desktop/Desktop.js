import React from "react";
import { arrayOf, shape, number, string, bool } from "prop-types";

import { Wrapper } from "../../styles";

import { translate } from "../../../../utils/i18n";
import formatNumber from "../../../../utils/formatNumber";
import Table from "../../../../components/common/Table/";
import DefaultContent from "../../../common/DefaultContent";
import { black30 } from "../../../../styles/settings";
import Icon from "../../../common/Icon";
import { formatDateToLocale } from "../../../../utils/formatDate";

const buildRows = rows =>
  rows.map(b => ({
    accountNumber: <div>{b.accountNumber}</div>,
    date: <div>{formatDateToLocale(b.date)}</div>,
    availableBalance: (
      <div>{formatNumber(b.availableBalance, { digits: 2 })}</div>
    ),
    blockedBalance: <div>{formatNumber(b.blockedBalance, { digits: 2 })}</div>,
    totalBalance: <div>{formatNumber(b.totalBalance, { digits: 2 })}</div>
  }));

function Desktop({ accounts, isEmpty, loading }) {
  return (
    <Wrapper data-test="Position_Accounts">
      {isEmpty ? (
        <DefaultContent
          data-test="Empty_Position"
          Icon={() => <Icon type="EmptyWallet" color={black30} />}
          primaryText={translate("NO_RECORDS")}
          secondaryTexts={[translate("NO_RECORD_WAS_FOUND")]}
        />
      ) : (
        <Table
          headers={[
            {
              title: translate("ACCOUNT_NUMBER"),
              field: "accountNumber",
              width: "25%"
            },
            { title: translate("REFERENCE_DATE"), field: "date" },
            {
              title: translate("AVAILABLE_BALANCE"),
              field: "availableBalance",
              currency: true,
              align: "right",
              width: "12%"
            },
            {
              title: translate("BLOCKED_BALANCE"),
              field: "blockedBalance",
              currency: true,
              align: "right"
            },
            {
              title: translate("TOTAL_BALANCE"),
              field: "totalBalance",
              currency: true,
              align: "right"
            },
            {
              title: "",
              field: "",
              width: "7%"
            }
          ]}
          data={buildRows(accounts)}
          shimmerLoading={{ rows: 10, loading }}
        />
      )}
    </Wrapper>
  );
}

Desktop.defaultProps = {
  accounts: []
};

Desktop.propTypes = {
  accounts: arrayOf(
    shape({
      id: number,
      accountNumber: string,
      date: string,
      totalBalance: number,
      availableBalance: number,
      blockedBalance: number
    })
  ),
  loading: bool.isRequired
};

export default Desktop;
