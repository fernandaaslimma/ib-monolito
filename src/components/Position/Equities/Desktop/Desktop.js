import React from "react";
import { arrayOf, shape, number, string, bool } from "prop-types";
import { Wrapper } from "../../styles";
import formatNumber from "../../../../utils/formatNumber";
import { isValueNullOrUndefined } from "../../../../utils/validations/input";
import formatDate from "../../../../utils/formatDate";

import DefaultContent from "../../../common/DefaultContent";
import { black30 } from "../../../../styles/settings";
import Icon from "../../../common/Icon";
import { UP, DEFAULT_VALUE } from "../../../../utils/constants";
import Table from "../../../../components/common/Table/";
import { translate } from "../../../../utils/i18n";

const buildRows = rows =>
  rows.map(b => ({
    name: <span>{b.name}</span>,

    date: <span>{formatDate(b.date)}</span>,
    quantity: (
      <span>
        {isValueNullOrUndefined(b.quantity)
          ? DEFAULT_VALUE
          : formatNumber(b.quantity, { digits: 0 })}
      </span>
    ),
    grossBalance: <span>{formatNumber(b.grossBalance, { digits: 2 })}</span>,
    portfolioShare: <span>{formatNumber(b.portfolioShare)}</span>
  }));

function Desktop({ isEmpty, equities, totalEquities, loading }) {
  return (
    <Wrapper>
      {isEmpty ? (
        <DefaultContent
          data-test="Empty_Position"
          Icon={() => <Icon type="EmptyWallet" color={black30} />}
          primaryText={translate("NO_RECORDS")}
          secondaryTexts={[translate("NO_RECORD_WAS_FOUND")]}
        />
      ) : (
        <Table
          headers={[
            {
              title: translate("NAME"),
              field: "name",
              width: "35%"
            },

            {
              title: translate("REFERENCE_DATE"),
              field: "date",
              width: "10%",
              align: "right"
            },
            {
              title: translate("QUANTITY"),
              field: "quantity",
              align: "right"
            },
            {
              title: translate("GROSS_BALANCE"),
              field: "grossBalance",
              currency: true,
              align: "right",
              total: true
            },
            {
              title: translate("PORTFOLIO"),
              field: "portfolioShare",
              percent: true,
              align: "right"
            }
          ]}
          data={buildRows(equities)}
          total={totalEquities}
          totalPosition={UP}
          shimmerLoading={{ rows: 10, loading }}
        />
      )}
    </Wrapper>
  );
}

Desktop.defaultProps = {
  equities: [],
  loading: false
};

Desktop.propTypes = {
  equities: arrayOf(
    shape({
      name: string,
      date: string,
      grossBalance: number,
      portfolioShare: number
    })
  ),
  loading: bool.isRequired,
  isEmpty: bool
};

export default Desktop;
