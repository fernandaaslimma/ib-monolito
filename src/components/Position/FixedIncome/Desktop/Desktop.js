import React from "react";
import { arrayOf, shape, number, string, bool } from "prop-types";
import { Wrapper, Name } from "../../styles";
import formatNumber from "../../../../utils/formatNumber";
import formatDate from "../../../../utils/formatDate";
import { isValueNullOrUndefined } from "../../../../utils/validations/input";

import DefaultContent from "../../../common/DefaultContent";
import { black30 } from "../../../../styles/settings";
import Icon from "../../../common/Icon";
import {
  UP,
  DEFAULT_VALUE,
  CREATE_TRANSACTION
} from "../../../../utils/constants";
import Table from "../../../../components/common/Table/";
import { translate } from "../../../../utils/i18n";
import { StickyWrapper } from "../../../Transfers/IndividualWireTransfer/styles";
import { AlertMessage, Button } from "react-bocombbm-components";
import { BtnWrapper, Separator } from "../../WithdrawLca/styles";
import { AlertWrapper } from "./styles";
import { checkIfHasAccess } from "../../../common/CanAccess/CanAccess";

const buildRows = rows =>
  rows.map(b => ({
    name: <Name>{b.name}</Name>,
    issuer: <span>{b.issuer}</span>,
    maturityDate: <span>{formatDate(b.maturityDate)}</span>,
    indexerRate: <span>{formatNumber(b.indexerRate)}</span>,
    indexer: <span>{b.indexer}</span>,
    date: <span>{formatDate(b.date)}</span>,

    quantity: (
      <span>
        {isValueNullOrUndefined(b.quantity)
          ? DEFAULT_VALUE
          : formatNumber(b.quantity, { digits: 2 })}
      </span>
    ),
    grossBalance: <span>{formatNumber(b.grossBalance, { digits: 2 })}</span>,
    netBalance: <span>{formatNumber(b.netBalance, { digits: 2 })}</span>,
    portfolioShare: <span>{formatNumber(b.portfolioShare)}</span>,
    incomeTaxBalance: (
      <span>{formatNumber(b.incomeTaxBalance, { digits: 2 })}</span>
    ),
    iofBalance: <span>{formatNumber(b.iofBalance, { digits: 2 })}</span>,
    grossResultBalance: (
      <span>{formatNumber(b.grossResultBalance, { digits: 2 })}</span>
    )
  }));

function Desktop({
  isEmpty,
  fixedIncome,
  totalFixedIncome,
  loading,
  pendingTransactions,
  totalLca,
  verifyOnGoingTransactions,
  userInfo
}) {
  return (
    <React.Fragment>
      <Wrapper>
        {isEmpty ? (
          <DefaultContent
            data-test="Empty_Position"
            Icon={() => <Icon type="EmptyWallet" color={black30} />}
            primaryText={translate("NO_RECORDS")}
            secondaryTexts={[translate("NO_RECORD_WAS_FOUND")]}
          />
        ) : (
          <Table
            headers={[
              { title: translate("NAME"), field: "name", width: "14%" },
              {
                title: translate("ISSUANCE"),
                field: "issuer",
                width: "15%"
              },
              {
                title: translate("MATURITY"),
                field: "maturityDate",
                align: "right"
              },
              {
                title: translate("RATE"),
                field: "indexerRate",
                percent: true,
                align: "right",
                width: "6%"
              },
              {
                title: translate("INDEX"),
                field: "indexer",
                width: "3%",
                align: "right"
              },
              {
                title: translate("REFERENCE_DATE"),
                field: "date",
                align: "right",
                width: "11%"
              },
              {
                title: translate("QUANTITY"),
                field: "quantity",
                align: "right"
              },
              {
                title: translate("GROSS_BALANCE"),
                field: "grossBalance",
                currency: true,
                align: "right",
                width: "12%",
                total: true
              },
              {
                title: translate("NET_BALANCE"),
                field: "netBalance",
                currency: true,
                align: "right",
                width: "11%",
                total: true
              },
              {
                title: translate("PORTFOLIO"),
                field: "portfolioShare",
                align: "right",
                percent: true,
                width: "7%",
                total: true
              },
              {
                title: translate("INCOME_TAX"),
                field: "incomeTaxBalance",
                currency: true,
                total: true,
                align: "right",
                width: "9.6%"
              },
              {
                title: translate("IOF"),
                field: "iofBalance",
                currency: true,
                total: true,
                align: "right",
                width: "6%"
              },
              {
                title: translate("GROSS_REVENUE"),
                field: "grossResultBalance",
                currency: true,
                align: "right",
                total: true,
                width: "15%"
              }
            ]}
            data={buildRows(fixedIncome)}
            total={totalFixedIncome}
            totalPosition={UP}
            shimmerLoading={{ rows: 10, loading }}
            pendingTransactions={pendingTransactions}
            userInfo={userInfo}
          />
        )}
      </Wrapper>
      {checkIfHasAccess(userInfo, [CREATE_TRANSACTION]) &&
      !loading &&
      (!totalLca || totalLca === 0) ? (
        <AlertWrapper data-test="alertMessageLCA">
          <AlertMessage
            icon="Attention"
            type="neutral"
            spacing={{
              top: "s",
              bottom: "s",
              right: "none",
              left: "none"
            }}
          >
            {translate("NO_LCA_WITH_LIQUIDITY")}
          </AlertMessage>
        </AlertWrapper>
      ) : null}

      {checkIfHasAccess(userInfo, [CREATE_TRANSACTION]) ? (
        <StickyWrapper>
          <Separator />
          <BtnWrapper>
            <Button
              dataTest="ButtonWithdrawLCA"
              disabled={!totalLca || totalLca === 0}
              spacing={{ top: "s", bottom: "s", right: "s", left: "s" }}
              onClick={() => verifyOnGoingTransactions()}
            >
              {translate("FIXED_INCOME_WITHDRAW_LCA")}
            </Button>
          </BtnWrapper>
        </StickyWrapper>
      ) : null}
    </React.Fragment>
  );
}

Desktop.defaultProps = {
  fixedIncome: [],
  loading: false,
  isEmpty: false
};

Desktop.propTypes = {
  fixedIncome: arrayOf(
    shape({
      name: string,
      date: string,
      issuer: string,
      maturityDate: string,
      grossBalance: number,
      netBalance: number,
      portfolioShare: number,
      indexerRate: number,
      indexer: string,
      incomeTaxBalance: number,
      iofBalance: number,
      quantity: number
    })
  ),
  loading: bool,
  isEmpty: bool
};

export default Desktop;
