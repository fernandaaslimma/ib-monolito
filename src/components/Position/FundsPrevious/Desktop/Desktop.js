import React from "react";
import { arrayOf, shape, number, string, bool } from "prop-types";
import { Wrapper } from "../../styles";
import formatNumber from "../../../../utils/formatNumber";
import formatDate from "../../../../utils/formatDate";
import { isValueNullOrUndefined } from "../../../../utils/validations/input";

import DefaultContent from "../../../common/DefaultContent";
import { black30 } from "../../../../styles/settings";
import Icon from "../../../common/Icon";
import { UP, DEFAULT_VALUE } from "../../../../utils/constants";
import Table from "../../../../components/common/Table/";
import { translate } from "../../../../utils/i18n";
import { Container } from "../../../../styles/grid";
import LinkCard from "../../../common/LinkCard";

const buildRowsFunds = rows =>
  rows.map(b => ({
    name: <span>{b.name}</span>,
    investmentDate: <span>{formatDate(b.investmentDate)}</span>,
    date: <span>{formatDate(b.date)}</span>,
    quantity: (
      <span>
        {isValueNullOrUndefined(b.quantity)
          ? DEFAULT_VALUE
          : formatNumber(b.quantity, { digits: 8 })}
      </span>
    ),
    grossBalance: <span>{formatNumber(b.grossBalance, { digits: 2 })}</span>,
    netBalance: <span>{formatNumber(b.netBalance, { digits: 2 })}</span>,
    incomeTaxBalance: (
      <span>{formatNumber(b.incomeTaxBalance, { digits: 2 })}</span>
    ),
    iofBalance: <span>{formatNumber(b.iofBalance, { digits: 2 })}</span>,
    grossResultBalance: (
      <span>{formatNumber(b.grossResultBalance, { digits: 2 })}</span>
    )
  }));

function Desktop({ isEmpty, funds, totalFunds, loading }) {
  return (
    <Wrapper>
      <Container>
        {isEmpty ? (
          <DefaultContent
            data-test="Empty_Position"
            Icon={() => <Icon type="EmptyWallet" color={black30} />}
            primaryText={translate("NO_RECORDS")}
            secondaryTexts={[translate("NO_RECORD_WAS_FOUND")]}
          />
        ) : (
          <Table
            headers={[
              {
                title: translate("NAME"),
                field: "name",
                width: "20%"
              },
              {
                title: translate("APPL_DATE"),
                field: "investmentDate",
                align: "right",
                widht: "12%"
              },
              {
                title: translate("REFERENCE_DATE"),
                field: "date",
                widht: "7%",
                align: "right"
              },
              {
                title: translate("QUANTITY"),
                field: "quantity",
                width: "10%",
                align: "right"
              },
              {
                title: translate("GROSS_BALANCE"),
                field: "grossBalance",
                currency: true,
                align: "right",
                widht: "12%",
                total: true
              },
              {
                title: translate("NET_BALANCE"),
                field: "netBalance",
                currency: true,
                align: "right",
                widht: "12%",
                total: true
              },
              {
                title: translate("INCOME_TAX"),
                field: "incomeTaxBalance",
                currency: true,
                align: "right",
                total: true
              },
              {
                title: translate("IOF"),
                field: "iofBalance",
                currency: true,
                align: "right",
                width: "5%"
              },
              {
                title: translate("GROSS_REVENUE"),
                field: "grossResultBalance",
                currency: true,
                align: "right",
                width: "14%",
                total: true
              }
            ]}
            data={buildRowsFunds(funds)}
            total={totalFunds}
            totalPosition={UP}
            shimmerLoading={{ rows: 10, loading }}
          />
        )}
        {!loading && (
          <LinkCard
            iconType="FiLoader"
            to="/investments/positions/funds"
            anchorText={translate("CLICK_TO_VIEW_FUNDS")}
            versionText={translate("WISH_TO_SEE_SIMPLIFIED_VERSION")}
            dataTest="LinkToSimplified"
            noSpan
            withUnderline
            fontSize={14}
          />
        )}
      </Container>
    </Wrapper>
  );
}

Desktop.defaultProps = {
  funds: [],
  loading: false
};

Desktop.propTypes = {
  funds: arrayOf(
    shape({
      name: string,
      date: string,
      investmentDate: string,
      iofBalance: number,
      grossResultBalance: number,
      grossBalance: number,
      incomeTaxBalance: number,
      portfolioShare: number,
      quantity: number
    })
  ),
  loading: bool,
  isEmpty: bool
};

export default Desktop;
