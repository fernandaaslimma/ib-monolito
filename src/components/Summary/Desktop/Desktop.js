import React, { Component } from "react";
import { string, arrayOf, shape, number } from "prop-types";

import { Wrapper } from "./styles";
import IndexTable from "./IndexTable";
import ConsolidatedPositionTable from "./ConsolidatedPositionTable";
import AssetsChart from "./AssetsChart";
import TransactionsTable from "./TransactionsTable";
import { Container, Row, Column } from "../../../styles/grid";
import { translate } from "../../../utils/i18n";
import { Button } from "react-bocombbm-components";
import { StickyWrapper } from "../Mobile/styles";
import { redirect } from "../../../utils/redirect";
import {
  INVESTMENT_FIXED_INCOME_ROLE,
  INVESTMENT_FUNDS_ROLE,
  INVESTMENT_PRODUCTS_LIST_URL
} from "../../../utils/constants";
import CanAccess from "../../common/CanAccess";
import PendingTransactions from "../../common/PendingTransactions";

class Desktop extends Component {
  render() {
    const {
      indexes,
      consolidatedPosition,
      consolidatedAssets,
      transactions,
      pendingTransactions,
      loading
    } = this.props;

    return (
      <Wrapper>
        <Container>
          <Row mb={20}>
            <Column width={1 / 2} ml={15} mr={2}>
              <AssetsChart
                assets={consolidatedPosition}
                title={translate("TYPE_OF_ASSETS_PERCENT_IN_PORTFOLIO")}
                icon="Portfolio"
                dataTest="Assets_Type"
                loading={loading}
              />
            </Column>
            <Column width={1 / 2} mr={15} ml={2}>
              <AssetsChart
                assets={consolidatedAssets}
                title={translate("ASSET_CLASS")}
                icon="Wallet"
                dataTest="Assets_Class"
                loading={loading}
              />
            </Column>
          </Row>
          {pendingTransactions && pendingTransactions.length > 0 && (
            <Row>
              <Column width={1 / 2} ml={15} mr={3}>
                <PendingTransactions
                  mode={"desktop"}
                  pendingTransactions={pendingTransactions}
                />
              </Column>
            </Row>
          )}
          <Row>
            <Column>
              <ConsolidatedPositionTable
                consolidatedPosition={consolidatedPosition}
              />
            </Column>
          </Row>
          <Row>
            <Column width={3 / 5} ml={15} mr={2}>
              <TransactionsTable transactions={transactions} />
            </Column>
            <Column width={2 / 5} ml={15} mr={2}>
              <IndexTable indexes={indexes} />
            </Column>
          </Row>
        </Container>
        <CanAccess
          userInfo={this.props.userInfo}
          roles={[INVESTMENT_FUNDS_ROLE, INVESTMENT_FIXED_INCOME_ROLE]}
          anyRole={true}
        >
          <StickyWrapper data-test="investButton">
            <Button
              onClick={() => redirect(INVESTMENT_PRODUCTS_LIST_URL)}
              disabled={false}
              spacing={{ top: "s", bottom: "s", right: "s", left: "s" }}
            >
              {translate("INVEST")}
            </Button>
          </StickyWrapper>
        </CanAccess>
      </Wrapper>
    );
  }
}

Desktop.defaultProps = {
  indexes: [],
  transactions: [],
  consolidatedPosition: [],
  consolidatedAssets: []
};

Desktop.propTypes = {
  indexes: arrayOf(
    shape({
      index: string,
      monthAcrrued: number,
      yearAccrued: number
    })
  ),
  consolidatedPosition: arrayOf(
    shape({
      name: string,
      netBalance: number,
      grossBalance: number,
      portfolioShare: number
    })
  ),
  consolidatedAssets: arrayOf(
    shape({
      name: string,
      netBalance: number,
      grossBalance: number,
      portfolioShare: number
    })
  ),
  transactions: arrayOf(
    shape({
      date: string,
      grossValue: number,
      type: string,
      assetName: string
    })
  )
};

export default Desktop;
