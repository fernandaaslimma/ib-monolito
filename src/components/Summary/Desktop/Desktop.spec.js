import React from "react";
import { shallow } from "enzyme";

import Desktop from "./Desktop";

const props = {
  indexes: [],
  consolidatedPosition: [
    {
      name: "",
      netBalance: 1,
      grossBalance: 1,
      portfolioShare: 1
    }
  ],
  consolidatedAssets: [
    {
      name: "",
      netBalance: 1,
      grossBalance: 2,
      portfolioShare: 3
    }
  ],
  transactions: [
    {
      date: "",
      grossValue: 1,
      type: "",
      assetName: ""
    }
  ]
};

it("should match snapshot", () => {
  expect(shallow(<Desktop {...props} />)).toMatchSnapshot();
});
