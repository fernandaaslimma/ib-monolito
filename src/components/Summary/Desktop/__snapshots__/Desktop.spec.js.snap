// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<styled.div>
  <Container>
    <Row
      mb={20}
    >
      <Column
        ml={15}
        mr={2}
        width={0.5}
      >
        <AssetsChart
          assets={
            Array [
              Object {
                "grossBalance": 1,
                "name": "",
                "netBalance": 1,
                "portfolioShare": 1,
              },
            ]
          }
          dataTest="Assets_Type"
          icon="Portfolio"
          title=""
        />
      </Column>
      <Column
        ml={2}
        mr={15}
        width={0.5}
      >
        <AssetsChart
          assets={
            Array [
              Object {
                "grossBalance": 2,
                "name": "",
                "netBalance": 1,
                "portfolioShare": 3,
              },
            ]
          }
          dataTest="Assets_Class"
          icon="Wallet"
          title=""
        />
      </Column>
    </Row>
    <Row>
      <Column>
        <ConnectWrapper
          consolidatedPosition={
            Array [
              Object {
                "grossBalance": 1,
                "name": "",
                "netBalance": 1,
                "portfolioShare": 1,
              },
            ]
          }
        />
      </Column>
    </Row>
    <Row>
      <Column
        ml={15}
        mr={2}
        width={0.6}
      >
        <TransactionsTable
          transactions={
            Array [
              Object {
                "assetName": "",
                "date": "",
                "grossValue": 1,
                "type": "",
              },
            ]
          }
        />
      </Column>
      <Column
        ml={15}
        mr={2}
        width={0.4}
      >
        <IndexTable
          indexes={Array []}
        />
      </Column>
    </Row>
  </Container>
  <ConnectWrapper
    anyRole={true}
    roles={
      Array [
        "GetInvestmentFunds",
        "GetFixedIncomeProducts",
      ]
    }
  >
    <styled.div
      data-test="investButton"
    >
      <Button
        dataTest="Button"
        disabled={false}
        onClick={[Function]}
        spacing={
          Object {
            "bottom": "s",
            "left": "s",
            "right": "s",
            "top": "s",
          }
        }
      />
    </styled.div>
  </ConnectWrapper>
</styled.div>
`;
